--- origsrc/php-5.2.6/ext/odbc/config.m4	2007-07-31 08:02:00.000000000 -0500
+++ src/php-5.2.6/ext/odbc/config.m4	2008-05-07 14:33:42.656250000 -0500
@@ -476,6 +476,26 @@
   fi
 fi
 
+if test -z "$ODBC_TYPE"; then
+PHP_ARG_WITH(msodbc,,
+[  --with-msodbc[=DIR]   Include MS-ODBC support [/usr]])
+
+  if test "$PHP_MSODBC" != "no"; then
+    AC_MSG_CHECKING(for MS-ODBC support)
+    if test "$PHP_MSODBC" = "yes"; then
+      PHP_MSODBC=/usr
+    fi
+    ODBC_INCDIR=$PHP_UNIXODBC/include
+    ODBC_LIBDIR=$PHP_UNIXODBC/$PHP_LIBDIR
+    ODBC_LFLAGS=-L$ODBC_LIBDIR
+    ODBC_INCLUDE=-I$ODBC_INCDIR
+    ODBC_LIBS=-lodbc32
+    ODBC_TYPE=msodbc
+    AC_DEFINE(HAVE_MSODBC,1,[ ])
+    AC_MSG_RESULT([$ext_output])
+  fi
+fi
+
 dnl
 dnl Extension setup
 dnl
--- origsrc/php-5.2.6/ext/odbc/php_odbc_includes.h	2007-12-31 01:20:09.000000000 -0600
+++ src/php-5.2.6/ext/odbc/php_odbc_includes.h	2008-05-07 14:33:42.656250000 -0500
@@ -195,6 +195,7 @@
 
 #else /* MS ODBC */
 
+#define ODBC_TYPE "MS ODBC"
 #define HAVE_SQL_EXTENDED_FETCH 1
 #include <WINDOWS.H>
 #include <sql.h>
--- origsrc/php-5.2.6/ext/pdo_odbc/config.m4	2007-07-09 07:35:44.000000000 -0500
+++ src/php-5.2.6/ext/pdo_odbc/config.m4	2008-05-07 14:33:42.671875000 -0500
@@ -84,6 +84,15 @@
         pdo_odbc_def_lib=odbcsdk
         ;;
 
+    ODBC32|odbc32)
+        pdo_odbc_def_libdir=/usr/lib/w32api
+        pdo_odbc_def_incdir=/usr/include/w32api
+        pdo_odbc_def_lib=odbc32
+        # w32api libs require the correct number of args, unlike AC_CHECK_LIB
+        ac_cv_lib_odbc32___SQLBindCol=yes
+        ac_cv_lib_odbc32___SQLAllocHandle=yes
+        ;;
+
     generic)
         pdo_odbc_def_lib="`echo $PHP_PDO_ODBC | cut -d, -f3`"
         pdo_odbc_def_ldflags="`echo $PHP_PDO_ODBC | cut -d, -f4`"
--- origsrc/php-5.2.6/ext/pdo_odbc/php_pdo_odbc_int.h	2007-12-31 01:20:10.000000000 -0600
+++ src/php-5.2.6/ext/pdo_odbc/php_pdo_odbc_int.h	2008-05-07 14:33:42.687500000 -0500
@@ -22,6 +22,11 @@
 # define PDO_ODBC_TYPE	"Win32"
 #endif
 
+#if defined(__CYGWIN__)
+# define WIN32_LEAN_AND_MEAN
+# include <windows.h>
+#endif
+
 #ifndef PDO_ODBC_TYPE
 # warning Please fix configure to give your ODBC libraries a name
 # define PDO_ODBC_TYPE	"Unknown"
@@ -43,6 +48,10 @@
 # include <iodbc.h>
 #endif
 
+#if HAVE_SQL_H
+# include <sql.h>
+#endif
+
 #if HAVE_SQLUNIX_H && !defined(PHP_WIN32)
 # include <sqlunix.h>
 #endif
@@ -55,10 +64,6 @@
 # include <sqlucode.h>
 #endif
 
-#if HAVE_SQL_H
-# include <sql.h>
-#endif
-
 #if HAVE_ISQL_H
 # include <isql.h>
 #endif
