--- origsrc/php-5.3.4/Makefile.global	2010-09-05 07:29:47.000000000 -0500
+++ src/php-5.3.4/Makefile.global	2010-12-29 23:23:07.000000000 -0600
@@ -39,8 +39,7 @@ install-modules: build-modules
 	@test -d modules && \
 	$(mkinstalldirs) $(INSTALL_ROOT)$(EXTENSION_DIR)
 	@echo "Installing shared extensions:     $(INSTALL_ROOT)$(EXTENSION_DIR)/"
-	@rm -f modules/*.la >/dev/null 2>&1
-	@$(INSTALL) modules/* $(INSTALL_ROOT)$(EXTENSION_DIR)
+	@$(INSTALL) modules/*.$(SHLIB_DL_SUFFIX_NAME) $(INSTALL_ROOT)$(EXTENSION_DIR)
 
 install-headers:
 	-@if test "$(INSTALL_HEADERS)"; then \
@@ -107,7 +106,7 @@ clean:
 	find . -name \*.gcno -o -name \*.gcda | xargs rm -f
 	find . -name \*.lo -o -name \*.o | xargs rm -f
 	find . -name \*.la -o -name \*.a | xargs rm -f 
-	find . -name \*.so | xargs rm -f
+	find . -name \*.$(SHLIB_DL_SUFFIX_NAME) | xargs rm -f
 	find . -name .libs -a -type d|xargs rm -rf
 	rm -f libphp$(PHP_MAJOR_VERSION).la $(SAPI_CLI_PATH) $(OVERALL_TARGET) modules/* libs/*
 
--- origsrc/php-5.3.4/acinclude.m4	2010-11-19 02:26:47.000000000 -0600
+++ src/php-5.3.4/acinclude.m4	2011-01-02 07:29:29.802495700 -0600
@@ -844,6 +844,10 @@ AC_DEFUN([PHP_SHARED_MODULE],[
       suffix=so
       link_cmd='$(LIBTOOL) --mode=link ifelse($4,,[$(CC)],[$(CXX)]) $(COMMON_FLAGS) $(CFLAGS_CLEAN) $(EXTRA_CFLAGS) $(LDFLAGS) -Wl,-G -o '$3'/$1.la -export-dynamic -avoid-version -prefer-pic -module -rpath $(phplibdir) $(EXTRA_LDFLAGS) $($2) $(translit($1,a-z_-,A-Z__)_SHARED_LIBADD) && mv -f '$3'/.libs/$1.so '$3'/$1.so'
       ;;
+    *cygwin*[)]
+      suffix=la
+      link_cmd='$(LIBTOOL) --mode=link ifelse($4,,[$(CC)],[$(CXX)]) $(COMMON_FLAGS) $(CFLAGS_CLEAN) $(EXTRA_CFLAGS) $(LDFLAGS) -o [$]@ -shared -export-dynamic -avoid-version -module -no-undefined -rpath $(phplibdir) $(EXTRA_LDFLAGS) $($2) ifelse($1, libphp5, , -lphp5) $(translit($1,a-z_-,A-Z__)_SHARED_LIBADD)'
+      ;;
     *netware*[)]
       suffix=nlm
       link_cmd='$(LIBTOOL) --mode=link ifelse($4,,[$(CC)],[$(CXX)]) $(COMMON_FLAGS) $(CFLAGS_CLEAN) $(EXTRA_CFLAGS) $(LDFLAGS) -o [$]@ -shared -export-dynamic -avoid-version -prefer-pic -module -rpath $(phplibdir) $(EXTRA_LDFLAGS) $($2) ifelse($1, php5lib, , -L$(top_builddir)/netware -lphp5lib) $(translit(ifelse($1, php5lib, $1, m4_substr($1, 3)),a-z_-,A-Z__)_SHARED_LIBADD)'
@@ -1990,6 +1994,10 @@ AC_DEFUN([PHP_SHLIB_SUFFIX_NAMES],[
    SHLIB_SUFFIX_NAME=dylib
    SHLIB_DL_SUFFIX_NAME=so
    ;;
+ *cygwin*[)]
+   SHLIB_SUFFIX_NAME=dll.a
+   SHLIB_DL_SUFFIX_NAME=dll
+   ;;
  esac
 ])
 
--- origsrc/php-5.3.4/configure.in	2010-12-08 15:38:51.000000000 -0600
+++ src/php-5.3.4/configure.in	2011-01-02 21:28:34.082290500 -0600
@@ -974,7 +974,7 @@ case $php_build_target in
         standard_libtool_flag='-prefer-non-pic'
         ;;
     esac
-    EXTRA_LDFLAGS="$EXTRA_LDFLAGS -avoid-version -module"
+    EXTRA_LDFLAGS="$EXTRA_LDFLAGS -avoid-version -module -no-undefined"
     ;;
 esac
 
@@ -1206,7 +1206,7 @@ case $host_alias in
 *)
   PHP_UNAME=`uname -a | xargs`
   AC_DEFINE_UNQUOTED(PHP_UNAME,"$PHP_UNAME",[uname -a output])
-  PHP_OS=`uname | xargs`
+  PHP_OS=`uname | xargs | sed -e 's/[[-_ ]].*//'`
   AC_DEFINE_UNQUOTED(PHP_OS,"$PHP_OS",[uname output])
   ;;
 esac
--- origsrc/php-5.3.4/sapi/apache/config.m4	2007-07-11 18:20:37.000000000 -0500
+++ src/php-5.3.4/sapi/apache/config.m4	2010-12-29 23:23:18.000000000 -0600
@@ -75,6 +75,10 @@ if test "$PHP_APXS" != "no"; then
     PHP_AIX_LDFLAGS="-Wl,-brtl"
     build_type=shared
     ;;
+  *cygwin*)
+    EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lhttpd"
+    build_type=shared
+    ;;
   *darwin*)
     MH_BUNDLE_FLAGS="-dynamic -twolevel_namespace -bundle -bundle_loader $APXS_HTTPD"
     PHP_SUBST(MH_BUNDLE_FLAGS)
--- origsrc/php-5.3.4/sapi/apache2handler/config.m4	2008-03-11 17:47:39.000000000 -0500
+++ src/php-5.3.4/sapi/apache2handler/config.m4	2010-12-29 23:23:18.000000000 -0600
@@ -49,7 +49,9 @@ if test "$PHP_APXS2" != "no"; then
     echo $APU_BINDIR/apu-config`
 
   APR_CFLAGS="`$APR_CONFIG --cppflags --includes`"
+  APR_LIBS="`$APR_CONFIG --ldflags --link-ld`"
   APU_CFLAGS="`$APU_CONFIG --includes`"
+  APU_LIBS="`$APU_CONFIG --ldflags --link-ld`"
 
   for flag in $APXS_CFLAGS; do
     case $flag in
@@ -58,6 +60,7 @@ if test "$PHP_APXS2" != "no"; then
   done
 
   APACHE_CFLAGS="$APACHE_CPPFLAGS -I$APXS_INCLUDEDIR $APR_CFLAGS $APU_CFLAGS"
+  EXTRA_LIBS="$APR_LIBS $APU_LIBS"
 
   # Test that we're trying to configure with apache 2.x
   PHP_AP_EXTRACT_VERSION($APXS_HTTPD)
@@ -87,17 +90,12 @@ if test "$PHP_APXS2" != "no"; then
     PHP_SELECT_SAPI(apache2handler, shared, mod_php5.c sapi_apache2.c apache_config.c php_functions.c, $APACHE_CFLAGS)
     INSTALL_IT="$INSTALL_IT $SAPI_LIBTOOL" 
     ;;
+  *cygwin*)
+    EXTRA_LDFLAGS="-shrext .so $EXTRA_LDFLAGS -L`$APXS -q LIBDIR` -lhttpd"
+    PHP_SELECT_SAPI(apache2handler, shared, mod_php5.c sapi_apache2.c apache_config.c php_functions.c, $APACHE_CFLAGS) 
+    INSTALL_IT="$INSTALL_IT $SAPI_LIBTOOL"
+    ;;
   *darwin*)
-    dnl When using bundles on Darwin, we must resolve all symbols.  However,
-    dnl the linker does not recursively look at the bundle loader and
-    dnl pull in its dependencies.  Therefore, we must pull in the APR
-    dnl and APR-util libraries.
-    if test -x "$APR_CONFIG"; then
-        MH_BUNDLE_FLAGS="`$APR_CONFIG --ldflags --link-ld --libs`"
-    fi
-    if test -x "$APU_CONFIG"; then
-        MH_BUNDLE_FLAGS="`$APU_CONFIG --ldflags --link-ld --libs` $MH_BUNDLE_FLAGS"
-    fi
     MH_BUNDLE_FLAGS="-bundle -bundle_loader $APXS_HTTPD $MH_BUNDLE_FLAGS"
     PHP_SUBST(MH_BUNDLE_FLAGS)
     PHP_SELECT_SAPI(apache2handler, bundle, mod_php5.c sapi_apache2.c apache_config.c php_functions.c, $APACHE_CFLAGS)
--- origsrc/php-5.3.4/sapi/cli/config.m4	2008-09-01 08:15:31.000000000 -0500
+++ src/php-5.3.4/sapi/cli/config.m4	2011-01-02 08:15:50.638543400 -0600
@@ -9,1 +9,1 @@ PHP_ARG_ENABLE(cli,,
-  SAPI_CLI_PATH=sapi/cli/php
+  SAPI_CLI_PATH=sapi/cli/php$EXEEXT
--- origsrc/php-5.3.4/sapi/fpm/config.m4	2010-11-26 07:46:15.000000000 -0600
+++ src/php-5.3.4/sapi/fpm/config.m4	2011-01-02 22:17:51.101730200 -0600
@@ -385,1 +385,1 @@ if test "$PHP_FPM" != "no"; then
-  SAPI_FPM_PATH=sapi/fpm/php-fpm
+  SAPI_FPM_PATH=sapi/fpm/php-fpm$EXEEXT
