--- origsrc/php-5.3.8/build/buildcheck.sh	2011-05-18 14:55:34.000000000 -0500
+++ src/php-5.3.8/build/buildcheck.sh	2011-09-20 21:03:41.369973800 -0500
@@ -44,7 +44,7 @@ echo "           to build PHP from SVN."
 exit 1
 fi
 
-if test "$1" = "2" -a "$2" -gt "59"; then
+if false; then
   echo "buildconf: You need autoconf 2.59 or lower to build this version of PHP."
   echo "           You are currently trying to use $ac_version"
   echo "           Most distros have separate autoconf 2.13 or 2.59 packages."
--- origsrc/php-5.3.4/configure.in	2010-12-08 15:38:51.000000000 -0600
+++ src/php-5.3.4/configure.in	2011-01-02 21:28:34.082290500 -0600
@@ -1,7 +1,7 @@
 ## $Id: configure.in 306090 2010-12-08 21:38:51Z johannes $ -*- autoconf -*-
 dnl ## Process this file with autoconf to produce a configure script.
 
-divert(1)
+divert(6001)
 
 dnl ## Diversion 1 is the autoconf + automake setup phase. We also
 dnl ## set the PHP version, deal with platform-specific compile
@@ -290,7 +290,7 @@ sinclude(TSRM/threads.m4)
 sinclude(TSRM/tsrm.m4)
 
 
-divert(2)
+divert(6002)
 
 dnl ## Diversion 2 is where we set PHP-specific options and come up
 dnl ## with reasonable default values for them. We check for pthreads here
@@ -329,7 +329,7 @@ if test "$enable_maintainer_zts" = "yes"
   PTHREADS_FLAGS
 fi
 
-divert(3)
+divert(6003)
 
 dnl ## In diversion 3 we check for compile-time options to the PHP
 dnl ## core and how to deal with different system dependencies. 
@@ -675,7 +675,7 @@ if test "x$php_crypt_r" = "x1"; then
   PHP_CRYPT_R_STYLE
 fi
 
-divert(4)
+divert(6004)
 
 dnl ## In diversion 4 we check user-configurable general settings.
 
@@ -916,7 +916,7 @@ else
   AC_MSG_RESULT([using system default])
 fi
 
-divert(5)
+divert(6005)
 
 dnl ## In diversion 5 we check which extensions should be compiled.
 dnl ## All of these are normally in the extension directories.
--- origsrc/php-5.3.4/ext/standard/config.m4	2010-06-17 05:22:03.000000000 -0500
+++ src/php-5.3.4/ext/standard/config.m4	2010-12-29 23:23:21.000000000 -0600
@@ -1,6 +1,6 @@
 dnl $Id: config.m4 300511 2010-06-17 10:22:03Z pajoye $ -*- autoconf -*-
 
-divert(3)dnl
+divert(6003)dnl
 
 dnl
 dnl Check if flush should be called explicitly after buffered io
@@ -342,7 +342,7 @@ dnl
 AC_CHECK_FUNCS(getcwd getwd asinh acosh atanh log1p hypot glob strfmon nice fpclass isinf isnan mempcpy strpncpy)
 AC_FUNC_FNMATCH	
 
-divert(5)dnl
+divert(6005)dnl
 
 dnl
 dnl Check if there is a support means of creating a new process
--- origsrc/php-5.3.4/scripts/phpize.m4	2010-11-02 04:58:08.000000000 -0500
+++ src/php-5.3.4/scripts/phpize.m4	2010-12-29 23:23:21.000000000 -0600
@@ -1,6 +1,6 @@
 dnl This file becomes configure.in for self-contained extensions.
 
-divert(1)
+divert(6001)
 
 AC_PREREQ(2.13)
 AC_INIT(config.m4)
