--- origsrc/php-5.3.4/ext/pspell/config.m4	2007-09-26 10:44:16.000000000 -0500
+++ src/php-5.3.4/ext/pspell/config.m4	2010-12-29 23:23:16.000000000 -0600
@@ -29,8 +29,6 @@ if test "$PHP_PSPELL" != "no"; then
 
 	PSPELL_LIBDIR=$PSPELL_DIR/$PHP_LIBDIR
 
-	PHP_ADD_LIBRARY_WITH_PATH(pspell, $PSPELL_LIBDIR, PSPELL_SHARED_LIBADD)
-
 	dnl Add -laspell to LIBS if it exists
 	PHP_CHECK_LIBRARY(aspell,new_aspell_config,
 	[
