https://bugs.php.net/bug.php?id=48820

--- origsrc/php-5.4.19/ext/posix/config.m4	2013-08-19 23:42:41.000000000 -0500
+++ src/php-5.4.19/ext/posix/config.m4	2013-08-23 13:34:53.452433200 -0500
@@ -13,8 +13,8 @@ if test "$PHP_POSIX" = "yes"; then
 
   AC_CHECK_FUNCS(seteuid setegid setsid getsid setpgid getpgid ctermid mkfifo mknod getrlimit getlogin getgroups makedev initgroups getpwuid_r getgrgid_r)
 
-  AC_MSG_CHECKING([for working ttyname_r() implementation])
-  AC_TRY_RUN([
+  AC_CACHE_CHECK([for working ttyname_r() implementation], ac_cv_func_ttyname_r_works, [
+    AC_TRY_RUN([
 #include <unistd.h>
 
 int main(int argc, char *argv[])
@@ -23,15 +23,21 @@ int main(int argc, char *argv[])
 
 	return ttyname_r(0, buf, 64) ? 1 : 0;
 }
-  ],[
-    AC_MSG_RESULT([yes])
-    AC_DEFINE(HAVE_TTYNAME_R, 1, [Whether you have a working ttyname_r])
-  ],[
-    AC_MSG_RESULT([no, posix_ttyname() will be thread-unsafe])
-  ], [
-    AC_MSG_RESULT([no, cannot detect working ttyname_r() when cross compiling. posix_ttyname() will be thread-unsafe])
+    ],[
+      ac_cv_func_ttyname_r_works=yes
+    ],[
+      ac_cv_func_ttyname_r_works=no
+    ], [
+      ac_cv_func_ttyname_r_works=no
+    ])
   ])
 
+  if test "$ac_cv_func_ttyname_r_works" = yes; then
+    AC_DEFINE(HAVE_TTYNAME_R, 1, [Whether you have a working ttyname_r])
+  else
+    AC_MSG_NOTICE([posix_ttyname() will be thread-unsafe])
+  fi
+
   AC_CACHE_CHECK([for utsname.domainname], ac_cv_have_utsname_domainname, [
     AC_TRY_COMPILE([
       #define _GNU_SOURCE
