--- origsrc/php-5.4.20/ext/gd/gd.c	2013-09-18 00:48:57.000000000 -0500
+++ src/php-5.4.20/ext/gd/gd.c	2013-10-17 15:31:10.833732700 -0500
@@ -56,6 +56,8 @@
 #if HAVE_LIBGD
 #if !HAVE_GD_BUNDLED
 # include "libgd/gd_compat.h"
+#else
+extern int overflow2(int, int);
 #endif
 
 
@@ -83,6 +85,9 @@ static void php_free_ps_enc(zend_rsrc_li
 #  include FT_FREETYPE_H
 # endif
 #endif
+#if defined(HAVE_GD_XPM) && defined(HAVE_GD_BUNDLED)
+# include <X11/xpm.h>
+#endif
 
 #ifndef M_PI
 #define M_PI 3.14159265358979323846
