--- origsrc/php-7.0.17/php.ini-production	2017-03-14 06:26:05.000000000 -0500
+++ src/php-7.0.17/php.ini-production	2017-03-16 11:18:06.913010400 -0500
@@ -985,7 +985,7 @@ cli_server.color = On
 
 ; Enables or disables JIT compilation of patterns. This requires the PCRE
 ; library to be compiled with JIT support.
-;pcre.jit=1
+pcre.jit=0
 
 [Pdo]
 ; Whether to pool ODBC connections. Can be one of "strict", "relaxed" or "off"
