https://cygwin.com/ml/cygwin/2017-05/msg00350.html

--- origsrc/php-7.1.9/Zend/zend_alloc.c	2017-08-30 11:13:20.000000000 -0500
+++ src/php-7.1.9/Zend/zend_alloc.c	2017-09-26 21:43:38.087650400 -0500
@@ -719,7 +719,7 @@ static void *zend_mm_chunk_alloc_int(siz
 		/* chunk has to be aligned */
 		zend_mm_munmap(ptr, size);
 		ptr = zend_mm_mmap(size + alignment - REAL_PAGE_SIZE);
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
 		offset = ZEND_MM_ALIGNED_OFFSET(ptr, alignment);
 		zend_mm_munmap(ptr, size + alignment - REAL_PAGE_SIZE);
 		ptr = zend_mm_mmap_fixed((void*)((char*)ptr + (alignment - offset)), size);
--- origsrc/php-7.1.13/Zend/zend_stream.c	2018-01-02 20:32:22.000000000 -0600
+++ src/php-7.1.13/Zend/zend_stream.c	2018-01-09 01:42:30.871472500 -0600
@@ -29,7 +29,9 @@
 #if HAVE_MMAP
 # if HAVE_UNISTD_H
 #  include <unistd.h>
-#  if defined(_SC_PAGESIZE)
+#  if defined(__CYGWIN__)
+#    define REAL_PAGE_SIZE 4096
+#  elif defined(_SC_PAGESIZE)
 #    define REAL_PAGE_SIZE sysconf(_SC_PAGESIZE);
 #  elif defined(_SC_PAGE_SIZE)
 #    define REAL_PAGE_SIZE sysconf(_SC_PAGE_SIZE);
--- origsrc/php-7.1.13/main/main.c	2018-01-02 20:32:22.000000000 -0600
+++ src/php-7.1.13/main/main.c	2018-01-09 01:42:57.670125500 -0600
@@ -97,7 +97,9 @@
 #if HAVE_MMAP || defined(PHP_WIN32)
 # if HAVE_UNISTD_H
 #  include <unistd.h>
-#  if defined(_SC_PAGESIZE)
+#  if defined(__CYGWIN__)
+#    define REAL_PAGE_SIZE 4096
+#  elif defined(_SC_PAGESIZE)
 #    define REAL_PAGE_SIZE sysconf(_SC_PAGESIZE);
 #  elif defined(_SC_PAGE_SIZE)
 #    define REAL_PAGE_SIZE sysconf(_SC_PAGE_SIZE);
