--- origsrc/php-7.2.5/ext/mbstring/config.m4	2018-04-24 10:09:55.000000000 -0500
+++ src/php-7.2.5/ext/mbstring/config.m4	2018-04-25 16:28:05.213732600 -0500
@@ -187,8 +187,8 @@ int main() { return foo(10, "", 3.14); }
         -L$PHP_ONIG/$PHP_LIBDIR
       ])
 
-      save_old_LDFLAGS=$LDFLAGS
-      PHP_EVAL_LIBLINE([$MBSTRING_SHARED_LIBADD], LDFLAGS)
+      save_old_LIBS=$LIBS
+      PHP_EVAL_LIBLINE([$MBSTRING_SHARED_LIBADD], LIBS)
       AC_MSG_CHECKING([if oniguruma has an invalid entry for KOI8 encoding])
       AC_TRY_LINK([
 #include <oniguruma.h>
@@ -200,7 +200,7 @@ return (int)(ONIG_ENCODING_KOI8 + 1);
         AC_MSG_RESULT([yes])
         AC_DEFINE([PHP_ONIG_BAD_KOI8_ENTRY], [1], [define to 1 if oniguruma has an invalid entry for KOI8 encoding])
       ])
-      LDFLAGS=$save_old_LDFLAGS
+      LIBS=$save_old_LIBS
     fi
 
     PHP_MBSTRING_ADD_CFLAG([-DONIG_ESCAPE_UCHAR_COLLISION=1])
