--- pear.orig//PEAR/Installer/Role/Ext.xml	2011-01-02 02:40:08.848229500 -0600
+++ pear/PEAR/Installer/Role/Ext.xml	2011-01-02 02:59:56.727900700 -0600
@@ -6,6 +6,6 @@
  <honorsbaseinstall>1</honorsbaseinstall>
  <unusualbaseinstall />
  <phpfile />
- <executable />
+ <executable>1</executable>
  <phpextension>1</phpextension>
  <config_vars />
--- pear.orig//PEAR/Installer.php	2011-01-02 02:40:08.879429600 -0600
+++ pear/PEAR/Installer.php	2011-01-02 03:50:06.940349200 -0600
@@ -441,7 +441,7 @@
             }
             // }}}
             // {{{ set file permissions
-            if (!OS_WINDOWS) {
+            if (!OS_WINDOWS && PHP_OS != 'CYGWIN') {
                 if ($atts['role'] == 'script') {
                     $mode = 0777 & ~(int)octdec($this->config->get('umask'));
                     $this->log(3, "+ chmod +x $dest_file");
@@ -656,7 +656,7 @@
 
             // }}}
             // {{{ set file permissions
-            if (!OS_WINDOWS) {
+            if (!OS_WINDOWS && PHP_OS != 'CYGWIN') {
                 if ($role->isExecutable()) {
                     $mode = 0777 & ~(int)octdec($this->config->get('umask'));
                     $this->log(3, "+ chmod +x $dest_file");
@@ -1587,7 +1587,7 @@
 
                 $this->log(3, "+ cp $ext[file] $copyto");
                 $this->addFileOperation('rename', array($ext['file'], $copyto));
-                if (!OS_WINDOWS) {
+                if (!OS_WINDOWS && PHP_OS != 'CYGWIN') {
                     $mode = 0666 & ~(int)octdec($this->config->get('umask'));
                     $this->addFileOperation('chmod', array($mode, $copyto));
                     if (!@chmod($copyto, $mode)) {
--- pear.orig//PEAR.php	2011-01-02 02:40:27.271861600 -0600
+++ pear/PEAR.php	2011-01-02 02:39:52.561801100 -0600
@@ -712,6 +712,8 @@
 
         if (OS_WINDOWS) {
             $suffix = '.dll';
+        } elseif (PHP_OS == 'CYGWIN') {
+            $suffix = '.dll';
         } elseif (PHP_OS == 'HP-UX') {
             $suffix = '.sl';
         } elseif (PHP_OS == 'AIX') {
